src/runtime/process.cpp

```diff
@@ -316,6 +320,20 @@ extern "C" LEAN_EXPORT obj_res lean_io_process_get_pid(obj_arg) {
     return lean_io_result_mk_ok(box_uint32(getpid()));
 }
 
+extern "C" LEAN_EXPORT obj_res lean_io_get_tid(obj_arg) {
+    uint64_t tid;
+#ifdef LEAN_WINDOWS
+    tid = GetCurrentThreadId();
+#elif defined(__APPLE__)
+    pthread_threadid_np(NULL, &tid);
```

Comment:
Optional suggestion: add a comment that this returns an error code but cannot fail in this case because it can only fail if the first agument is non-null or the second argument is null.

Reply:
Thanks for pointing that out, let's be consistent here then!